public/app/plugins/datasource/grafana-azure-monitor-datasource/components/LogsQueryEditor/QueryField.tsx

```diff
@@ -0,0 +1,32 @@
+import { CodeEditor } from '@grafana/ui';
+import React, { useCallback } from 'react';
+import { AzureQueryEditorFieldProps } from '../../types';
+
+const QueryField: React.FC<AzureQueryEditorFieldProps> = ({ query, onQueryChange }) => {
+  const onChange = useCallback(
+    (newQuery: string) => {
+      onQueryChange({
+        ...query,
+        azureLogAnalytics: {
+          ...query.azureLogAnalytics,
+          query: newQuery,
+        },
+      });
+    },
+    [onQueryChange, query]
+  );
+
+  return (
+    <CodeEditor
+      value={query.azureLogAnalytics.query}
+      language="kql"
```

Comment:
Is there a language definition for KQL? I can't seem to find it [here](https://github.com/microsoft/monaco-languages/tree/main/src).

Reply:
no not yet. i'll remove this for now :)

Reply:
oh. language is mandatory. I'll leave it there, it just won't do anything yet.